lib/bootloader_setup.pm

```diff
@@ -181,25 +181,46 @@ Boot the default kernel recovery mode (selected in the "Advanced options ..."):
     boot_grub_item(2, 2);
 
 =cut
+# private function
+sub grub_key_down {
+    if (get_var('S390_ZKVM')) {
+        type_line_svirt '$\'\\e\'[B', use_printf => 1;
```

Comment:
```suggestion
        type_line_svirt(q($'\\e'[B), use_printf => 1);
```

Reply:
This is quite complicated as it also takes https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html into the game. 
What do you think about

https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html

```suggestion
        type_line_svirt(q(-en '\\e[B'));
```
Or we make a proper escaping in type_line_svirt() with printf.

```
clemix@linux-7k5u:~> echo -en '\e[B' | xxd
00000000: 1b5b 42                                  .[B
clemix@linux-7k5u:~> printf '\e[B' | xxd
00000000: 1b5b 42                                  .[B
clemix@linux-7k5u:~> printf $'\\e'[B | xxd
00000000: 1b5b 42                                  .[B
clemix@linux-7k5u:~> 
```

Reply:
It might be even better to add these escape codes to `os-autoinst/consoles/serial_screen.pm`.

Reply:
Well I usually prefer `printf` for escape sequences as it's it's portable (unlike `echo`). But ok, I'll take `type_line_svirt(q(-en '\\e[B'));` as it allows to remove support for `printf` (previous commit).

Reply:
@mdoucha I don't understand. Where would you add it there?

Reply:
@mdoucha agree (I was thinking about it as well while implementing it), but first I'd merge this, to get support immediately. I'll do that separately (I'll add up key as well).

Reply:
> @mdoucha I don't understand. Where would you add it there?

@cfconrad Here: https://github.com/os-autoinst/os-autoinst/blob/master/consoles/serial_screen.pm#L60
Then the entire `type_line_svirt` workaround can be removed.

Reply:
In general I agree.
I'm just confused by this redirection `> $pty` inside of `type_line_svirt()` which looks for me a odd and I'm not sure if it is the same if we do a write on the specific fd.

Reply:
Yep, it's confusing, but that's the way it works also without svirt serial console. If you look into history, it has been used long before svirt serial console. Thus adding key up and down will need to be implemented also on non-svirt-serial to be working on both setups.
\+ I also hacked this into test instead of adding key up/down into consoles because there are other problems with svirt (poo#54578, poo#45863) and I'd like finally to concentrate on tests itself than tooling.